### PR TITLE
Reuse envoy interface 

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/xds/fake.go [hunk 1]
```diff
@@ -51,6 +51,9 @@ import (
 	"istio.io/istio/pkg/test"
 )
 
+// DiscoveryClient is a client interface for XDS.
+type DiscoveryClient = discovery.AggregatedDiscoveryService_StreamAggregatedResourcesClient
```
Comment (howardjohn) [inline pilot/pkg/xds/fake.go:55]: can you keep this in ads.go? its more generic than fake

File: pilot/pkg/xds/ads.go [hunk 2]
```diff
@@ -185,14 +173,14 @@ func (s *DiscoveryServer) receive(con *Connection, reqChannel chan *discovery.Di
 // handles 'push' requests and close - the code will eventually call the 'push' code, and it needs more mutex
 // protection. Original code avoided the mutexes by doing both 'push' and 'process requests' in same thread.
 func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *Connection) error {
-
-	if !s.preProcessRequest(con.proxy, req) {
+	if req.TypeUrl == v3.HealthInfoType {
```
Comment (howardjohn) [inline pilot/pkg/xds/ads.go:176]: I like the old code better, this leaks low level details into processRequest

File: pilot/pkg/xds/workloadentry.go [hunk 3]
```diff
@@ -145,9 +145,6 @@ func (sg *InternalGen) QueueUnregisterWorkload(proxy *model.Proxy) {
 // UpdateWorkloadEntryHealth updates the associated WorkloadEntries health status
 // based on the corresponding health check performed by istio-agent.
 func (sg *InternalGen) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthEvent) {
-	if !features.WorkloadEntryHealthChecks {
```
Comment (howardjohn) [inline pilot/pkg/xds/workloadentry.go:148]: why is this removed?

File: pilot/pkg/xds/fake.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/fake.go:55]: sure

File: pilot/pkg/xds/workloadentry.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/workloadentry.go:148]: It is moved before calling this function to prevent unnecessary HealthEvent convert

File: pilot/pkg/xds/ads.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go:176]: I would argue that TypeUrl should be a high level stuff, it's more like a switch case.

And also I donot like the return value `should continue`, lol

File: pilot/pkg/xds/ads.go [hunk 4]
```diff
@@ -185,14 +176,14 @@ func (s *DiscoveryServer) receive(con *Connection, reqChannel chan *discovery.Di
 // handles 'push' requests and close - the code will eventually call the 'push' code, and it needs more mutex
 // protection. Original code avoided the mutexes by doing both 'push' and 'process requests' in same thread.
 func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *Connection) error {
-
-	if !s.preProcessRequest(con.proxy, req) {
+	if req.TypeUrl == v3.HealthInfoType && features.WorkloadEntryHealthChecks {
```
Comment (howardjohn) [inline pilot/pkg/xds/ads.go:179]: I think my comment got hidden in rebase, but I think the old code here was better, as it abstracts the details of health check

Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go]: @howardjohn Reverted

**Review by howardjohn — APPROVED** [review]
